src/sentry/api/invite_helper.py

```diff
@@ -225,8 +224,9 @@ def accept_invite(self, user: User | None = None) -> OrganizationMember | None:
 
     def _needs_2fa(self) -> bool:
         org_requires_2fa = self.organization.flags.require_2fa.is_set
-        user_has_2fa = Authenticator.objects.user_has_2fa(self.request.user.id)
-        return org_requires_2fa and not user_has_2fa
+        return org_requires_2fa and (
+            self.request.user.id is None or not self.request.user.has_2fa()
```

Comment:
Should you use `request.user.is_authenticated()` instead of checking for the user's id?

Reply:
Yup, I'll make this change,.